hamilton/driver.py

```diff
@@ -47,6 +49,19 @@ def __init__(self, config: Dict[str, Any], *modules: ModuleType, adapter: base.H
         self.graph = graph.FunctionGraph(*modules, config=config, adapter=adapter)
         self.adapter = adapter
 
+    def _node_is_required_by_anything(self, node_: node.Node) -> bool:
+        """Tells us if a node is required by any other node
+
```

Comment:
Required here is a bit of a confusing term, i.e. if there's a link it's required in some sense. 

This is probably better to describe this funciton:
```suggestion
        """Checks dependencies on this node and determines if at least one requires it.
        
        Nodes can be optional, i.e. the function parameter has a default value, we want to check that
        of the nodes the depend on this one, at least one of them requires it, i.e. the parameter is not optional.
        
```